emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/DynamoDBClient.java

```diff
@@ -459,13 +459,7 @@ protected AwsCredentialsProvider getAwsCredentialsProvider(Configuration conf) {
     // initialized
     String providerClass = conf.get(DynamoDBConstants.CUSTOM_CREDENTIALS_PROVIDER_CONF);
     if (!Strings.isNullOrEmpty(providerClass)) {
-      try {
-        providersList.add(
-            (AwsCredentialsProvider) ReflectionUtils.newInstance(Class.forName(providerClass), conf)
-        );
-      } catch (ClassNotFoundException e) {
-        throw new RuntimeException("Custom AWSCredentialsProvider not found: " + providerClass, e);
-      }
+      providersList.add(DynamoDBUtil.loadAwsCredentialsProvider(providerClass, conf));
```

Comment:
Not related to this change but can we add DefaultProvider Chain instead of InstanceProfileCredProvider as it's not something thats supported by EMR Serverless ?

Reply:
yeah i was planning on doing this as a follow-up given it needs additional testing